blueprint/core/blueprint_FileWatcher.h

```diff
@@ -0,0 +1,97 @@
+/*
+  ==============================================================================
+
+    blueprint_FileWatcher.h
+    Created: 10 Oct 2020 3:08:39pm
+
+  ==============================================================================
+*/
+
+#pragma once
+
+
+namespace blueprint
+{
+
+    //==============================================================================
+    /** Helper class which watches files for changes and triggers a user supplied
+     *  callback in thte event of a file change.
+     */
+    class FileWatcher : private juce::Timer
+    {
+    public:
+        using FileChangedCallback = std::function<void(void)>;
+
+        explicit FileWatcher (FileChangedCallback && callback)
+            : onFileChanged(std::move(callback)) {}
+
+        void start() { startTimer(50); }
+        void stop() { stopTimer(); }
+
+        void watch (const juce::File& f)
+        {
+            jassert(f.existsAsFile());
```

Comment:
Super minor but maybe we should assert `watch` is called on message thread just to avoid any daft timer disasters with people trying to load things on background threads etc.